gprofiler/python.py

```diff
@@ -260,9 +281,9 @@ def determine_profiler_class(storage_dir: str, stop_event: Event):
 
 
 def get_python_profiler(
-    frequency: int, duration: int, stop_event: Event, storage_dir: str
+    frequency: int, duration: int, stop_event: Event, storage_dir: str, reinitialize_profiler: Callable[[], None]
 ) -> Union[PythonEbpfProfiler, PySpyProfiler]:
     global _profiler_class
     if _profiler_class is None:
         _profiler_class = determine_profiler_class(storage_dir, stop_event)
-    return _profiler_class(frequency, duration, stop_event, storage_dir)
+    return _profiler_class(frequency, duration, stop_event, storage_dir, reinitialize_profiler)
```

Comment:
`reinitialize_profiler` isn't specific to a profiler class. I think it's better to store it as a module global.

Reply:
Okie